src/swap/components/Swap.tsx

```diff
@@ -8,15 +8,89 @@ import { getSwapQuote } from '../core/getSwapQuote';
 import { isSwapError } from '../core/isSwapError';
 import { TextTitle3 } from '../../internal/text';
 import { formatTokenAmount } from '../../utils/formatTokenAmount';
-import type { SwapError, SwapReact } from '../types';
+import { useBalance, useReadContract } from 'wagmi';
+import { getTokenBalances } from '../core/getTokenBalances';
+import { erc20Abi } from 'viem';
+import type { Address } from 'viem';
+import type { SwapError, SwapQuoteLoadingState, SwapReact } from '../types';
 import type { Token } from '../../token';
+import type { UseBalanceReturnType, UseReadContractReturnType } from 'wagmi';
 
 export function Swap({ address, children, title = 'Swap' }: SwapReact) {
   const [error, setError] = useState<SwapError>();
   const [fromAmount, setFromAmount] = useState('');
   const [fromToken, setFromToken] = useState<Token>();
   const [toAmount, setToAmount] = useState('');
   const [toToken, setToToken] = useState<Token>();
+  const [swapQuoteLoadingState, setSwapQuoteLoadingState] =
+    useState<SwapQuoteLoadingState>({
+      isFromQuoteLoading: false,
+      isToQuoteLoading: false,
+    });
+
+  // returns ETH balance
+  const ethBalanceResponse: UseBalanceReturnType = useBalance({
+    address,
+  });
+
+  // returns erc20 token balance
+  const fromTokenBalanceResponse: UseReadContractReturnType = useReadContract({
```

Comment:
We should probably abstract this into his own Hook. As we are going to reuse it soon.

Reply:
agreed